asyncify.mjs

```diff
@@ -48,10 +36,31 @@ function proxyGet(obj, transform) {
   });
 }
 
-class Asyncify {
-  constructor() {
+export class Asyncify {
```

Comment:
I'm not sure how comfortable I am with exporting this class - its methods are definitely not for public consumption, they were meant to be an implementation detail whereas now they will become part of public API.

Instead, could you perhaps add 3rd optional parameter to `Instance`, `instantiate` and `instantiateStreaming` and pass that parameter directly to this constructor?

Reply:
Happy to do that but will no longer be able to use for my use case.  An example would be a library that operates on an already instantated WebAssembly module, that automatically asyncifies it before calling one of the exports:
eg. https://nodejs.org/api/wasi.html#wasistartinstance using `Asyncify` internally

Reply:
Hmm why not asyncify in advance? Doing it as part of instantiation makes a bit more sense semantically.

Reply:
Ah right, and another part of the reason is that you need to wrap all the _imports_ too. Those can be provided only before the instantiation happened.

Reply:
Not sure where you landed on this, but I'd prefer to expose the Asyncify class and add any "advanced" settings there vs adding additional arguments to the 3 instantiate APIs.  My rationale:
- it works for my [use case](https://github.com/cloudflare/workers-wasi/blob/main/src/index.ts) :)
- keeps change smaller
- more flexible for advanced users, in general I would argue anyone configuring the data region explicitly should be comfortable dealing with Asyncify internals

Reply:
I think there's a difference between dealing with Asyncify internals and committing to API stability of previously internal Asyncify class. If there's a specific function you need, I'd still lean towards extracting that specific API TBH so that the overall surface to maintain is smaller :)